synapse/lib/stormlib/scrape.py

```diff
@@ -154,9 +155,12 @@ async def _methGenMatches(self, text, pattern, fangs=None, flags=regex.IGNORECAS
         if fangs:
             _fangs = {src: dst for (src, dst) in fangs}
             _fangre = s_scrape.genFangRegex(_fangs)
-            scrape_text, offsets = s_scrape.refang_text2(text, re=_fangre, fangs=_fangs)
+            if len(text) < s_scrape.SCRAPE_SPAWN_LENGTH:
+                scrape_text, offsets = s_scrape.refang_text2(text, re=_fangre, fangs=_fangs)
+            else:
+                scrape_text, offsets = await s_coro.semafork(s_scrape.refang_text2, text, re=_fangre, fangs=_fangs)
```

Comment:
If we are using the forkpool here I think you can avoid the length check since the overhead is so low

Reply:
updated

Reply:
does refang take long enough to need a forkpool?

Reply:
it takes between 1-1.5sec on the 25MB test file i've been using so it seemed like a good idea

Reply:
per discussion, reverted since the length comparison is a very cheap constant comparison and worth avoiding the pickling cost for using the forkpool